src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,66 @@
+import { useContext, useState } from "react";
+import { UsernameContext } from "../contexts/Username";
+import { postCommentByUsername } from "../api";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCheckCircle } from "@fortawesome/free-solid-svg-icons";
+
+function CommentAdder({ article_id }) {
+	const { username } = useContext(UsernameContext);
+	const [commentInput, setCommentInput] = useState("");
+	const [isPosting, setIsPosting] = useState(false);
+	const [error, setError] = useState("");
+	const [postSuccess, setPostSuccess] = useState(false);
+
+	function handleChange(event) {
+		setError("");
+		setCommentInput(event.target.value);
+	}
+
+	function handleClick(event) {
+		event.preventDefault();
+		setIsPosting(true);
+		postCommentByUsername(article_id, username, commentInput)
+			.then((response) => {
+				setCommentInput("");
+				setPostSuccess(true);
+				setTimeout(() => setPostSuccess(false), 4000);
+				setTimeout(() => setIsPosting(false), 4000);
```

Comment:
You may consider reducing the timeout on this or handling this in a different way (maybe with a loading state in the same way we handle article load) as it just says loading for quite a while, even though we get the success message straight away - it's a bit confusing for the user! We are waiting 4 seconds after the request has been successful and the comment has been added to the db to get rid of the loading message (we are inside the .then() of the postComment suggesting by this point, loading is over)